### PR TITLE
override marker added to missing methods.

### PR DESCRIPTION
**Pull request by devnexen**
few casts left.

### CODE REVIEW

File: source/utils/Helper.cpp
Comment (jbeich) [inline source/utils/Helper.cpp]: `__LP32__` doesn't exist and `!defined(__ILP32__)` would break x86_64 again. `size_t` is `uint64_t` aka `unsigned long` on Linux/FreeBSD x86_64.

File: source/utils/Helper.cpp
Comment (jbeich) [inline source/utils/Helper.cpp]: Unused on non-OpenBSD and may break on build on 32bit platforms where `uint64_t` is `unsigned long long` but `toString(uint64_t d)` is already defined earlier.

File: source/utils/Helper.cpp
Comment (devnexen) [reply source/utils/Helper.cpp]: The only alternative I see is then creating special rules for OpenBSD and general rules for the rest. __LP32__ does not really exist just realise clang create this macro possibly.

Comment (jbeich) [reply source/utils/Helper.cpp]: If build broke because `size_t` != `uint64_t` on 64bit platform then `toString(size_t d)` is useful.

File: source/utils/Helper.cpp
Comment (jbeich) [inline source/utils/Helper.cpp]: Better move `toString(size_t d)` undef `#if defined(__OpenBSD__) && defined(__LP64__)` then drop `#ifdef __LP64__`.

**Review by jbeich — CHANGES_REQUESTED** [review]
Before 8568c7d14fb8 the code worked fine **as is** on both x86 and x86_64. I'm not sure why you're trying to complicate ifdefs.

File: source/utils/Helper.cpp
Comment (devnexen) [reply source/utils/Helper.cpp]: I propose we merge your PR then will apply your changes to this one keeping the override fixes ... What do you think ?

Comment (jbeich) [reply source/utils/Helper.cpp]: Sorry, I had a typo in my comment (edited). OpenBSD probably only needs `toString(size_t d)` only on 64bit platforms.

Comment (jbeich) [reply source/utils/Helper.cpp]: If my #1286 introduces a regression on OpenBSD then it shouldn't be merged.

Comment (devnexen) [reply source/utils/Helper.cpp]: Yes ideally the toString might be better as generics but that would require too much changes I fear.

Comment (devnexen) [reply source/utils/Helper.cpp]: I ve tested my branch on FreeBSD and OpenBSD 64 bits so far by the way.

Comment (jbeich) [reply source/utils/Helper.cpp]: i386 or x86 (32bit) is not dead yet. It has Tier1 support on FreeBSD.
```c++
source/gamemap/TileContainer.cpp:400:52: error: call to 'toString' is ambiguous
            OD_LOG_ERR("Unknown neighbor index=" + Helper::toString(i));
                                                   ^~~~~~~~~~~~~~~~
source/utils/LogManager.h:32:155: note: expanded from macro 'OD_LOG_ERR'
#define OD_LOG_ERR(_message)                      LogManager::getSingleton().logMessage(LogMessageLevel::CRITICAL, __FILE__, __LINE__, (std::string("") + _message))
                                                                                                                                                          ^~~~~~~~
source/utils/Helper.h:131:17: note: candidate function
    std::string toString(float f, unsigned short precision = 6);
                ^
source/utils/Helper.h:132:17: note: candidate function
    std::string toString(double d, unsigned short precision = 6);
                ^
source/utils/Helper.h:133:17: note: candidate function
    std::string toString(int8_t d);
                ^
source/utils/Helper.h:134:17: note: candidate function
    std::string toString(uint8_t d);
                ^
source/utils/Helper.h:135:17: note: candidate function
    std::string toString(int16_t d);
                ^
source/utils/Helper.h:136:17: note: candidate function
    std::string toString(uint16_t d);
                ^
source/utils/Helper.h:137:17: note: candidate function
    std::string toString(int32_t d);
                ^
source/utils/Helper.h:138:17: note: candidate function
    std::string toString(int64_t d);
                ^
source/utils/Helper.h:144:17: note: candidate function
    std::string toString(uint64_t d);
                ^
```

Comment (devnexen) [reply source/utils/Helper.cpp]: Ok so since your PR for Helpers change seem to work (tested locally your changes with my own branch) might be good to merge yours then will rebase this one. What do you think ?

Comment (jbeich) [reply source/utils/Helper.cpp]: If the changes here won't touch `toString` helpers simply rebase against  `development` branch. No need to wait for #1286 to be merged. Obviously, history would look better if you squash reverts, split unrelated bits into separate commits and document compiler error messages.

Comment (devnexen) [reply source/utils/Helper.cpp]: OK will try later in the day. Cheers.

**Comment by devnexen** [comment]
I hope it looks a bit better now. Sorry for the overly long review. Regards.

**Comment by stavrossk** [comment]
Hi, thanks for the great game. Does the 32-bit Windows version work on 64-bit Windows?

**Comment by hwoarangmy** [comment]
Hi, thanks for the PR. I will have a look ASAP

File: source/network/ODServer.cpp
Comment (hwoarangmy) [inline source/network/ODServer.cpp:1510]: I guess it would have been better to define indexSk as a size_t instead of casting every time there is a compare

**Comment by hwoarangmy** [comment]
The PR looks good. Merging

File: source/network/ODServer.cpp
Comment (hwoarangmy) [reply source/network/ODServer.cpp:1510]: Actually, it seems like Travis build fails because of that. I'm fixing it

Comment (devnexen) [reply source/network/ODServer.cpp:1510]: Yes I pushed a change too late few minutes after you merged :-) but feel free.

Comment (hwoarangmy) [reply source/network/ODServer.cpp:1510]: I already created the PR, thanks ^^

**Pull request merged**
